api/app/Models/Community.php

```diff
@@ -36,6 +36,21 @@ class Community extends Model
 
     public $guarded = [];
 
+    /**
+     * Boot function for using with Community
+     *
+     * @return void
+     */
+    protected static function boot()
+    {
+        parent::boot();
+        static::created(function (Community $community) {
+            $community->team()->firstOrCreate([], [
+                'name' => 'community-'.$community->id,
+            ]);
+        });
+    }
+
```

Comment:
@tristan-orourke 

This was also a thing in one of my PRs, I believe you intentionally did not want community->team relations created by default?

Reply:
I was helping Matt with this one and it is impossible to add members until the team is created since it was to insert a team based role when the team did not exist. 

Unless he was supposed to use the `addCommunityAdmins` method in some new mutation instead of using the existing `updateUserRoles` mutation :confused:

Reply:
Okay, for my PR it was to do with factory logic, but it probably should be consistent in both cases?

![image](https://github.com/user-attachments/assets/59035211-bbe9-4497-a463-765e2a584c1b)

Reply:
Yes, I've been thinking that the safest way to do this was to rely on the `addCommunityAdmins` and `addCommunityRecruiters` methods.

Reply:
My reasoning is that because of the way we used a polymorphic relationship with teams, the database can't guarantee the Community has a team at all. Therefore, its best to assume it might not, and check any time we need to (which these methods encapsulate).

I totally may have missed some reason this is a bad idea though 😅

Reply:
Hmm. I guess I figured that at this point we'd be using an UpdateUserRoles mutation class, instead of still using Lighthouse directives 🤔

Reply:
I suppose we're keeping this then?

Reply:
Yes, keeping the boot method.